### ISSUE TITLE
Support request handling for multiple requests?

### ISSUE THREAD

**Issue by rubenduiveman** [knee-cola/jest-mock-axios#94]
Hi!

First of all, I'm very happily using this library, so thanks! There is one thing I'd like to be able to do: I would like to set up handing requests beforehand instead of having to `mockResponse` or `mockError` resolve every request. I would like to have this to allow testing background requests that are scheduled.

I see two possible options to implement this:

1. integrate an event emitter library like [nanoevents](https://github.com/ai/nanoevents ) and emit an event when a new request is added. I could then subscribe to that event and fire a  `mockResponse` or `mockError` - whatever I see fit.
2. provide someting like `mockAxios.handler((req) => {  ... })`  that gets triggered every time a new request is added.

What do you think?

Cheers!
R

**Comment by rubenduiveman** [comment]
I currently work around this by using `setInterval`:

```ts
setInterval(() => {
  const req = mockAxios.lastReqGet();
  if (!req) { return; }
  mockAxios.mockResponse({ data: "OK" }, req);
}, 50);
```

**Comment by kingjan1999** [comment]
Hi,

sorry for the late response. I believe your second idea (using a callback function) should be doable quite easily without changing the way this library works significantly (basically this would mean invoking the callback [here](https://github.com/knee-cola/jest-mock-axios/blob/master/lib/mock-axios.ts#L64)). I'll take a look if I can implement this in the next few days, though I'm quite busy right now. Otherwise feel free to submit a PR :)

**Comment by rubenduiveman** [comment]
I [created a PR](https://github.com/knee-cola/jest-mock-axios/pull/95). Let me know what you think!

**Comment by kingjan1999** [comment]
Closed via #95
